src/autofill-utils.js

```diff
@@ -566,6 +566,34 @@ function getActiveElement (root = document) {
     return innerActiveElement
 }
 
+/**
+ * Takes a root element, creates a treewalker and finds all shadow elements that match the selector
+ * @param {HTMLElement|HTMLFormElement} root
+ * @param {string} selector
+ * @returns {Element[]}
+ */
+function findEnclosedElements (root, selector) {
+    // Check if there are any normal elements that match the selector
+    const elements = [...root.querySelectorAll(selector)]
+    if (elements.length > 0) {
+        return elements
+    }
```

Comment:
Small optimization here - check for the length before you convert into an array (avoids iteration + creation of additional empty array)

```suggestion
    const elements = root.querySelectorAll(selector)
    if (elements.length > 0) {
        return Array.from(elements)
    }
```